File: scratch/rewrite_cards.js
const fs = require('fs');

const raw = fs.readFileSync('public/data/cards.json', 'utf8');
const cards = JSON.parse(raw);

cards.forEach(card => {
    // We already patched characters in DB, let's also patch it in JSON so they are in sync
    if (card.name && card.name.includes(']')) {
        const parts = card.name.split(']');
        const charName = parts[1].trim();
        if (charName) {
            card.characters = [charName];
        }
    }

    if (card.skills) {
        card.skills.forEach(skill => {
            if (skill.trigger_conditions && Array.isArray(skill.trigger_conditions)) {
                const newConds = {};
                skill.trigger_conditions.forEach(tc => {
                    if (tc.variable) {
                        if (!newConds[tc.variable]) {
                            newConds[tc.variable] = [];
                        }
                        if (!newConds[tc.variable].includes(tc.value)) {
                            newConds[tc.variable].push(tc.value);
                        }
                    }
                });
                
                // Simplify array of 1 to just the value for cleaner UI? 
                // Wait, MongoDB handles arrays transparently for equality checks.
                // e.g. `{ "running_style": [2, 3] }` matches `running_style == 2`.
                // Actually, if we leave it as an array, the UI's dropdown for values might show "[2,3]" instead of "2".
                // Our UI extracts primitive values. If it's an array, it recurses.
                // Wait! If it recurses, the path will just be `skills.trigger_conditions.running_style`.
                // BUT it will extract `2` and `3` as separate options! This is perfect!
                
                // Let's replace it
                skill.trigger_conditions = newConds;
            }
        });
    }
});

fs.writeFileSync('public/data/cards.json', JSON.stringify(cards, null, 2));
console.log('Rewrote cards.json');

File: scratch/rewrite_cards.py
import json

with open('public/data/cards.json', 'r') as f:
    cards = json.load(f)

for card in cards:
    if 'name' in card and ']' in card['name']:
        char_name = card['name'].split(']', 1)[1].strip()
        if char_name:
            card['characters'] = [char_name]

    if 'skills' in card:
        for skill in card['skills']:
            if 'trigger_conditions' in skill and isinstance(skill['trigger_conditions'], list):
                new_conds = {}
                is_obj_array = False
                for tc in skill['trigger_conditions']:
                    if isinstance(tc, dict) and 'variable' in tc:
                        is_obj_array = True
                        var = tc['variable']
                        val = tc['value']
                        if var not in new_conds:
                            new_conds[var] = []
                        if val not in new_conds[var]:
                            new_conds[var].append(val)
                
                if is_obj_array:
                    # Simplify single-item arrays
                    for k, v in new_conds.items():
                        if len(v) == 1:
                            new_conds[k] = v[0]
                    skill['trigger_conditions'] = new_conds

with open('public/data/cards.json', 'w') as f:
    json.dump(cards, f, indent=2)

print('Rewrote cards.json')
